Revisions to XTQL tutorial, made into a guide for docs + tests

Akeboshiwind opened the pull request:
- Renamed `dev/xtql/README.md` `XTQL Walkthrough` and made into a guide
- Minimally tweaked some wording
- Added tests
- Use [asciidoc include statements](https://docs.asciidoctor.org/asciidoc/latest/directives/include-tagged-regions/) to include the code directly from the tests
  - NOTE: paths are relative to the root of the docs subfolder, as that's where asciidoctor is run from

Caveats:
- Commented out some parts of the tests:
  - `pull` for JSON
  - `UPDATE` with subquery: https://github.com/xtdb/xtdb/issues/3050
  - `ASSERT`
- Uses yaml string to store json & sql
  - This is because of a few constraints:
        1. The format needs to support multiline strings (json doesn't)
        2. The examples need to be strings (json will later be passed to the http api as `xtql.json` will soon be removed)
        3. The strings themselves need to contain comments for asciidoc so as to not include the containing format
            (i.e. if you have the comments outside of the string it will include at least the surrounding quotes)

Closes xtdb/devrel#20

Akeboshiwind committed:
fix(docs): update xtql tutorial based on tests

Akeboshiwind committed:
test(docs): add tests for xtql tutorial

For context the one at: dev/xtql/README.adoc

FiV0 committed:
cleanup tutorial-test

FiV0 commented on src/test/clojure/xtdb/tutorial_test.clj:
```diff
@@ -14,69 +14,133 @@
+(def documents
+  [[:put :documents {:xt/id "doc-id", :version 1}]
+   [:put :documents {:xt/id "doc-id2", :version 1}]])
+
```
If you only use data once then inline it in the submit call directly. Otherwise this is good.

FiV0 commented on src/test/clojure/xtdb/tutorial_test.clj:
```diff
@@ -14,69 +14,133 @@
-          '(from :users [{:xt/id "james"} first-name last-name]))
+    (t/is (= []
+             (xt/q tu/*node*
+                   '(from :users [{:xt/id "james"} first-name last-name]))))
```
Can you replace this by something that returns data. (`:ivan` for example).

Akeboshiwind committed:
docs: use asciidoc includes to reference actual tests

Akeboshiwind committed:
Merge branch '2.x' into xtql-tutorial-revisions

Akeboshiwind committed:
fix: inc -> + 1

Akeboshiwind committed:
fix: comment out update w/ subquery

Akeboshiwind committed:
refactor: remove commented code

Akeboshiwind committed:
fix: move to storing examples in strings in a yaml file

I chose a yaml file for because we have a couple of constraints:
1. The format needs to support multiline strings
2. The examples need to be strings (json will later be passed to the
   http api)
3. The strings themselves need to contain comments for asciidoc so as to
   not include the containing format (i.e. if you have the comments
   outside of the string it will include at least the quotes)

Akeboshiwind committed:
fix: rename walkthorugh, tweak wording & link

Akeboshiwind committed:
feat: add walkthough to guides section

FiV0 commented on src/test/clojure/xtdb/tutorial_test.clj:
```diff
@@ -0,0 +1,736 @@
+(defn sql-example [name]
+  (get examples name))
+
+;; TODO: Inline some
```
Still TODO?

FiV0 commented on src/test/clojure/xtdb/tutorial_test.clj:
```diff
@@ -0,0 +1,736 @@
+    ;; end::DML-Insert-xtql[]
+    ,)
+
+  ;; TODO: Test valid time
```
Still TODO?

FiV0 commented on src/test/clojure/xtdb/tutorial_test.clj:
```diff
@@ -0,0 +1,736 @@
+  (xt/submit-tx tu/*node*
+    [(xt/sql-op (sql-example "DML-Insert-sql"))])
+
+  ;; TODO: Test valid time
```
Idem

Akeboshiwind committed:
refactor: remove extra test

Akeboshiwind committed:
refactor: use :key-fn to test that all query outputs are the same

Akeboshiwind committed:
refactor: inline reused data

Akeboshiwind committed:
fix: add bitemporal tests

Akeboshiwind committed:
refactor: rename tutorial_test to xtql_walkthrough_test
